crates/swc_plugin_proxy/src/source_map/plugin_source_map_proxy.rs

```diff
@@ -1 +1,32 @@
-pub struct PluginSourceMapProxy {}
+#[cfg(feature = "plugin-mode")]
+use swc_common::{BytePos, Loc};
+
+#[cfg(feature = "plugin-mode")]
+#[cfg_attr(not(target_arch = "wasm32"), allow(unused))]
+use crate::memory_interop::read_returned_result_from_host_fallible;
+
+#[cfg(target_arch = "wasm32")]
+extern "C" {
+    fn __lookup_char_pos_source_map_proxy(byte_pos: u32, allocated_ret_ptr: i32) -> i32;
+}
+
+#[cfg(feature = "plugin-mode")]
+#[derive(Debug, Copy, Clone)]
+pub struct PluginSourceMapProxy;
+
+/// Subset of SourceMap interface supported in plugin.
+/// Unlike `Comments`, this does not fully implement `SourceMap`.
+#[cfg(feature = "plugin-mode")]
+impl PluginSourceMapProxy {
```

Comment:
As commented unlike `Comment` this doesn't implement `SourceMap` fully. There are some looks like redundant for the plugin, such as `new` / `load_*`, etcs. 

While we expand this and eventually all of impl seems needed can change this trivially.